src/openrct2/scenario/ObjectiveGoalGroup.h

```diff
@@ -0,0 +1,123 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2022 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../common.h"
+#include "ObjectiveGoal.h"
+
+class ObjectiveGoal;
+using ObjectiveGoalPtr = std::shared_ptr<ObjectiveGoal>;
```

Comment:
Why are you using shared ptr in general this should be advised. Objectives don't exactly make sense needing a shared ptr

Reply:
Then what do you suggest I use instead? I can't just put them in as is, as the different objectives have different memory sizes. Normal pointers? But then, how to deal with memory management and preventing from going out of scope, or staying scope when they're not needed anymore?

Reply:
@MaxMallon can you not use `unique_ptr`?